Add tronald-dump-skill

krisgesling opened the pull request:
**Tronald Dump Skill**: _Kris Gesling_

## Description:

**Ask Tronald Dump, President of the USA, about a range of topics. Quotes courtesy of the tronalddump.io API.**

## Checklist:
  - [x] Used [Meta Editor](http://rawgit.com/MycroftAI/mycroft-skills/master/meta_editor.html) to generate the skill README
  - [x] Skill has been tested and works
  - [x] Read and verified approval process - https://mycroft.ai/documentation/skills/skills-acceptance-process (Can require a community member vouching for skill)
  - [x] .gitmodules has been updated with the following:
  
```
+[submodule "NAME OF YOUR SKILL"]
+	path = name-of-your-skill
+	url = URL.FOR.YOUR.SKILL.git
```

 - [ ] README.md has been updated with your skill phrase and description.
^^ No longer required? ^^

KathyReid commented:
## Test of `tronald-dump` Skill

* Platform: Linux Ubuntu 18.04 LTS 
* Device version: Version 18.2.2
* Who: @kathyreid / @kathy-mycroft
* Datestamp: Thu 11 Oct 23:45:56 AEDT 2018
* Language and dialect of tester: Native English speaker, Australian dialect

## Standard checks (non-code checks)

* Check that the Skill installs using voice commands. Mycroft will get the user to confirm which Skill should be installed if there is ambiguity in Skill names - such as duplicate names. If possible, name the Skill so that there is minimal duplication and/or conflict. You should also verify that the Skill name can be verbally pronounced by speaking the Skill name into the Mycroft command line several times, and reading the resulting transcriptions. Suggest alternative Skill names if it is difficult to verbally pronounce the Skill name.

> The Skill is not in `mycroft-skills` yet, so it won't respond to voice commands. Using `msm` install instead. 

```
(.venv) kathyreid@kathyreid-Oryx-Pro:~/mycroft-core$ msm install https://github.com/krisgesling/tronald-dump-skill
INFO - Downloading skill: https://github.com/krisgesling/tronald-dump-skill
INFO - Successfully installed tronald-dump-skill
```

> Check that STT picks up the skill name for voice install

* Check the `dialog` directory to ensure that from a voice user interface perspective the dialogs read well. It is a good idea to run the `dialog` phrases through [mimic](https://mycroft.ai/documentation/mimic/). Sometimes the `dialog` files will need a small tweak such as a space between words, or extra vowel sounds, to sound realistic. Sometimes words don't render as expected and alternative wording should be used. Alway play every `dialog` phrase on the command line by doing `say <your dialog phrase here>` so that you can check how the `dialog` is spoken. Some of the tricks you might need to use include separating words by a space - such as `sub sonic` instead of `subsonic`, or `broad cast` instead of `broadcast`.

> install tronald dump                                                                         
 >> tronald dump .krisgesling has already been                                                
    installed

> This means that STT picks up the keywords OK

* Check the **Intents** defined in the `vocab` diectory. Determine if the **Intents** are likely to conflict with those from **Skills** that are installed by default, and if so, suggest alternatives if this makes sense, or at least advise the Skill Author that there are likely to be `Intent Collisions`. [Integration Test Runner](https://mycroft.ai/documentation/skills/automatic-testing/) can also assist with this.

_NOTE: Skills that are installed by default vary by `enclosure` and can be found in the `mycroft-skills` repo in the files called `DEFAULT.enclosure`. For example, the default Skills for Picroft are in a file called `DEFAULT.picroft`._

> The Intents are quite specific; unlikely to conflict. 

* Check the regular expressions defined in the `regex` directory - do they appear sensible? [Integration Test Runner](https://mycroft.ai/documentation/skills/automatic-testing/) can also assist with this.

> `regex` not used, Skipping check

* Check README for completeness - does it follow the README template and include all the relevant sections such as Intents, known issues, dependencies and so on?

> README is lovely

* Check `requirements.txt` and `requirements.sh` - are the required dependencies listed? If `requirements.sh` is used, is some form of conditional processing done to match against multiple distros? Often Skill Authors will add requirements.txt using only an “library=1.x.x” instead of “library >=1.x.x”. Check to make sure that here is an equal or greater than in the requirements to help future-proof the Skill, unless a _specific_ version is needed. 

> no requirements used, Skipping check

* Check `settingsmeta.json` - are the settings well laid out? Does the placeholder text make sense?

> The default `settingsmeta.json` has been left in place, even though the settings are not used. This means that the user sees this in home.mycroft.ai: 

![screenshot from 2018-10-12 00-09-14](https://user-images.githubusercontent.com/114158/46806521-82acc180-cdb3-11e8-937f-f5e30e400c22.png)

_ACTION: The `settingsmeta.json` file needs to be removed please._

## Functional tests

* "What does Tronald Dump think about Obama?"
> President Obama thinks the nation is not as divided as people think. He is living in a world of the make believe!

* "What does Tronald Dump have to say about the economy?"
> How can Crooked Hillary put her husband in charge of the economy when he was responsible for NAFTA, the worst economic deal in U.S. history? 

* "Tronald what do you think about guns?"
> Crooked Hillary said that I want guns brought into the school classroom. Wrong!

* "Tronald tell me about Hillary?"
> In Crooked Hillary's telepromter speech yesterday, she made up things that I said or believe but have no basis in fact. Not honest!

=> Functionally working very nicely. 
_Side note: oh, if only we could train a mimic2 voice on this ..._

## Other information

One of the Tests in your Skill is failing; 

---
=================================== FAILURES ===================================
 TestCase.test_skill[/opt/mycroft/skills/tronald-dump-skill.krisgesling-test103] 

self = <test.integrationtests.skills.discover_tests.TestCase object at 0x7fe998124b00>
skill = '/opt/mycroft/skills/tronald-dump-skill.krisgesling'
test = ('/opt/mycroft/skills/tronald-dump-skill.krisgesling/test/intent/dump.tronald.intent.1.intent.json', None)

    @pytest.mark.parametrize("skill,test", sum([
        [(skill, test) for test in tests[skill]]
        for skill in tests.keys()
        if not skill_dir or abspath(skill).startswith(abspath(skill_dir))
        ], []))
    def test_skill(self, skill, test):
        example, test_env = test
        if test_env:
            assert test_env.test_runner(skill, example, emitter, loader)
        else:
            t = SkillTest(skill, example, emitter)
            if not t.run(loader):
>               assert False, "Failure: " + t.failure_msg
E               AssertionError: Failure: ['match', 'utterance', 'Crooked Hillary wants to take your 2nd Amendment rights away. Will guns be taken from her heavily armed Secret Service detail? Maybe not!*']
E               assert False

test/integrationtests/skills/discover_tests.py:95: AssertionError
----------------------------- Captured stdout call -----------------------------
``

I took a quick look at this and I *think* the reason is that the Intent doesn't always bring back the *same* response from the tronalddump.io service - so the utterance is not predictable. I'd recommend changing it so that it just checks to see if an utterance *exists*. 

## Python checks of the code

* Inline documentation - is the code documented in line?

> No but it's a simple Skill

* Libraries - does the Skill include the correct libraries? Does it use too many libraries or dependencies?

> Doesn't use any deps

* Callback check - do we need to do a check for the callback error that was in the Spotify Skill?

> No, skipping check

* Error handling - are there any specific checks we make for error handling or graceful degradation?

> handles errors gracefully

* Automatic tests - do automatic tests of the **Skill** pass?

> No, more on this above

* Skill Settings using Python Booleans - Skill Settings were updated in August 2018 to use Python Booleans instead of text Boolean values used in Python 2.4. [This is a good example of the change to check for](https://github.com/MycroftAI/skill-date-time/pull/32/files?utf8=%E2%9C%93&diff=split)

> Skill Settings not used, skipping check

## Summary

* `skillsmeta.json` should be removed
* Test needs to be fixed because value returned from tronalddump.io varies 

Other than that, this was very nicely done!

If you can fix these things up I can merge :+1:

krisgesling commented:
Thanks for the review and nice comments Kathy, I have addressed the two issues.   

With the failing test I tried to still test the response with:
`"expected_response": ".*Social Service.*"`   
and on my local machine it was passing consistently, but Jenkins still failed. So I've removed the field altogether which I assume still requires the utterance to exist.

Sorry for the slow response too, just had an excellent (and sweaty) 10 days out in the Kimberley to catch the final days before the wet season sets in.

KathyReid commented:
Many thanks @krisgesling, appreciate you fixing those up.
